content/en/docs/setup/version-skew-policy.md

```diff
@@ -0,0 +1,128 @@
+
+Example:
+
+* `kube-apiserver` instances are at **1.13** and **1.12**
+* `kube-controller-manager` and `kube-scheduler` communicate with a load balancer that can route to any `kube-apiserver` instance
+* `kube-controller-manager` and `kube-scheduler` are supported at **1.12** (**1.13** is not supported because that would be newer than the `kube-apiserver` instance at version **1.12**)
+
+### `kubectl`
+
+`kubectl` is supported within one minor version (older or newer) of `kube-apiserver`.
+
+Example:
+
+* `kube-apiserver` is at **1.13**
+* `kubectl` is supported at **1.14**, **1.13**, and **1.12**
+
+Note that if version skew exists between `kube-apiserver` instances in a multi-server cluster, this narrows the supported `kubectl` versions.
+
+Example:
+
+* `kube-apiserver` instances are at **1.13** and **1.12**
+* `kubectl` is supported at **1.13** and **1.12** (other versions would be more than one minor version skewed from one of the `kube-apiserver` components)
+
+## Supported component upgrade order
+
+The supported version skew between components has implications on the order in which components must be upgraded.
+This section describes the order in which components must be upgraded to transition an existing cluster from version **1.n** to version **1.(n+1)**.
+
+### `kube-apiserver`
+
+Pre-requisites:
+
+* In a multi-server cluster, other `kube-apiserver` instances are at **1.n** or **1.(n+1)** (this ensures maximum skew of 1 minor version between the oldest and newest `kube-apiserver` instance)
+* The `kube-controller-manager` and `kube-scheduler` instances that communicate with this server are at version **1.n** (this ensures they are not newer than the existing API server version, and are within 1 minor version of the new API server version)
+* `kubelet` instances on all nodes are at version **1.n** or **1.(n-1)** (this ensures they are not newer than the existing API server version, and are within 2 minor versions of the new API server version)
+* Registered admission webhooks are capable of understanding all the versions of REST resources they are registered for (including any new versions available in **1.(n+1)**)
+
+Upgrade `kube-apiserver` to **1.(n+1)**
+
+### `kube-controller-manager` and `kube-scheduler`
+
+Pre-requisites:
+
+* The `kube-apiserver` instances these components communicate with are at **1.(n+1)** (in multi-server clusters in which these control plane components communicate with any `kube-apiserver` instance in the cluster, all `kube-apiserver` instances must be upgraded before upgrading these components)
+
+Upgrade `kube-controller-manager` and `kube-scheduler` to **1.(n+1)**
+
+### `kubelet`
+
+Pre-requisites:
+
+* The `kube-apiserver` instances the `kubelet` communicates with are at **1.(n+1)**
+
+Optionally upgrade `kubelet` instances to **1.(n+1)** (or they can be left at **1.n** or **1.(n-1)**)
```

Comment:
How to upgrade a cluster with a a kube-apiserver at 1.n and a kubelet 1.n-1 ? We might want to bump kubelet first then upgrade the apiserver, which isn’t the canonical upgrade path. WDYT @liggitt ?

Reply:
Is it acceptable to skip versions when you upgrade kubelets?

Reply:
Agree we should explicitly state that the supported way to upgrade kubelets is one version at a time

Reply:
> yastij:
> How to upgrade a cluster with a kube-apiserver at 1.n+1 and a kubelet 1.n-1?

This doc describes upgrading a cluster from 1.n to 1.n+1. Before upgrading an apiserver, all kubelets must be within one minor version of the current apiserver version:

> `kubelet` instances on all nodes are at version **1.n** or **1.(n-1)** (this ensures they are not newer than the existing API server version, and are within 2 minor versions of the new API server version)

added the questions about skip-level upgrades for kubelets (as well as whether draining pods is required as part of upgrade to follow up questions)